### PR TITLE
chore: #12184 MVP split engine file

### LINKED ISSUE #12184: Establish pattern for splitting Engine.ts file
### What is this about?

The `Engine.ts` file is a large ~2500 LOC file that holds the logic for controller initialization and is used in many places in the app. The goal here is to establish a pattern for splitting out the file so that it is:
- Easier to manage
- Better testability
- Reduce non-relevant codeowner reviews (saves time on PRs)
- More scalable solution

### Scenario

_No response_

### Design

_No response_

### Technical Details

- Since controllers tend to be dependent on one another, the imports from abstracted files can be functions that take  dependent controllers as args to ensure initialization order is respected. For example, if `TokenListController` is dependent on `NetworkController`, the abstracted TokenListController file could look like `initializeTokenListController(networkController: NetworkController)`

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- App behavior should remain the same
- Abstract at least one area as POV that can be applied to other pieces later on

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

- Slack thread - https://consensys.slack.com/archives/C02U025CVU4/p1730814155345939

### PR DESCRIPTION
**Pull request by Cal-L**
## **Description**

This PR establishes a pattern for splitting out controllers from the main `Engine.ts` file, which provides the following benefits
- Reduce friction regarding code ownership especially during reviews (this is the main purpose of this effort)
- Easier to manage, test, and handling controller initialization failures

The POC here applies to the `AccountsController` initialization logic, which has been mostly abstracted out into `AccountsController` directory. This pattern should be applicable to other controllers in the Engine file as well.

This pattern also enables us to redirect users to vault recovery in the event of controller initialization failures and should reduce the chances of users being "bricked". These changes may be introduced in future interations

## **Related issues**

Fixes: #12184 

## **Manual testing steps**

The changes should not affect app behavior
- Run the app as usual
- Ensure that everything behaves as usual (Auth, dapp connections, navigation, etc)

### **After**

App still runs normally

https://github.com/user-attachments/assets/dc7fc017-1a26-4681-b84a-e04c63952281

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/Engine/controllers/AccountsController/utils.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  initialState?: AccountsControllerState;
+}): AccountsController => {
+  let accountsController = {} as AccountsController;
+
+  try {
+    const accountsControllerMessenger: AccountsControllerMessenger =
+      messenger.getRestricted({
+        name: 'AccountsController',
+        allowedEvents: [
+          'SnapController:stateChange',
```
Comment (Cal-L) [inline app/core/Engine/controllers/AccountsController/utils.ts:38]: Once these controller actions and events are abstracted, these will not longer be hardcoded strings. They will use the imported constants instead

**Review by Cal-L — COMMENTED** [review]
We are not exporting via index for AccountsController here since it exposes the risk of circular dependencies (caused by action/event dependencies)

**Comment by Cal-L** [comment]
Skipping sonar failure (Engine.ts) since this PR just removes code from the file

**Review by tommasini — APPROVED** [review]

**Review by gambinish — APPROVED** [review]

**Review by EtherWizard33 — APPROVED** [review]
LGTM

**Pull request merged**